### DISCUSSION TITLE
Export `useGtag` function

### DISCUSSION THREAD

**Discussion by StijnMaenhautCZ** [johannschopplich/nuxt-gtag#95] [Q&A]
### Describe the feature

Currently, we're importing the `useGtag` function as follows:

```ts
import { useGtag } from '#imports';
```

As we put all our components in Storybook as well, and Storybook can't really cope with `#imports` (it errors out), I was wondering if you could make the `useGtag` function available through the package exports itself?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Can you think of other implementations of this feature?

### Final checks

- [X] Check existing [issues](https://github.com/johannschopplich/nuxt-gtag/issues).

**Comment by johannschopplich** [comment]
Unfortunately this is not possible. Technically, `useGtag` could be exported individually, but without Nuxt context it wouldn't make much sense and would fail anyway. Also, it would be bad practice to add this explicit export since the composable only works in a Nuxt context.

I assume you have the same issues in other components with other composables, don't you? I recommend the official Nuxt Storybook module to allow for auto-imports. Or do you use it already?

**Comment by StijnMaenhautCZ** [comment]
Yes, I'm using the [@nuxtjs/storybook](https://www.npmjs.com/package/@nuxtjs/storybook) module. The error is the following:

<img width="553" alt="SCR-20241128-prdy" src="https://github.com/user-attachments/assets/752c7643-79b3-4674-8611-2f8fb4010edb">

Would it be a solution to provide the Nuxt context to `useGtag` as an argument?

**Comment by johannschopplich** [comment]
I don't know how the Nuxt context is set up within Storybook. Do other auto-imports work? Can you remove the explicit import from `#imports` and rely on auto-imports instead?

**Comment by StijnMaenhautCZ** [comment]
Thanks for your quick replies! 

I have been able to bypass all other `#imports` by using a "direct import". This is the only package left 😃 

I have removed the import line, but the issue remains.

**Reply by johannschopplich** [reply]
I have converted the issue to a discussion, since the problem you encounter isn't necessarily tied to this module, but applies to all Nuxt composables. 

I think your question is better off in the Nuxt Discord channel, maybe there's one for Storybook integration as well? Feel free to share your findings here for other users. That would be much appreciated!

**Reply by StijnMaenhautCZ** [reply]
Will do and report back to this conversation if the answer is found.
